src/native/libs/System.Native/pal_interfaceaddresses.c

```diff
@@ -100,12 +101,66 @@ static inline uint8_t mask2prefix(uint8_t* mask, int length)
     return len;
 }
 
+#if !HAVE_IFADDRS && TARGET_ANDROID
+// This structure is exactly the same as struct ifaddrs defined in ifaddrs.h but since the header
+// might not be available (e.g., in bionics used in Android before API 24) we need to mirror it here
```

Comment:
```suggestion
// might not be available (e.g., in bionic used in Android before API 24) we need to mirror it here
```

Reply:
Does this actually happen? I thought the NDK headers define the struct regardless of the Android API version.

Reply:
Yes, the header file is missing until Android 7 (see https://cs.android.com/android/platform/superproject/+/android-6.0.1_r81:bionic/libc/include/, https://cs.android.com/android/platform/superproject/+/android-7.0.0_r1:bionic/libc/include/)

Reply:
Hmm, I think you're right. We don't need to define it since we're building the code with NDK r23c.